CHANGELOG.md

```diff
@@ -6,6 +6,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased] - TBD
 
+### Deprecated
+
+- Item Search methods `get_all_items` and `get_all_items_as_dict` are now deprecated.
+  These have been deprecated because they have the potential to perform a large number
+  of requests to the server and instantiate a large number of objects in memory.
+  To a user, this is only visible as a large delay in the method call and/or the
+  exhaustion of all available memory. The iterator methods `get_items` or
+  `get_itemcollections` should be used instead.
```

Comment:
Do we want to specify when the methods will be removed, or leave that unbounded?

Reply:
I would be in favor of saying it's going to be removed in 0.5.

Reply:
👍🏽, but I'd like to have @matthewhanson's ok as well if possible just b/c this is the first time we're deprecating in here (I think).

Reply:
also as a reminder, the semver convention is that minor versions prior to 1.0 are the same as major version after.

Reply:
I weasled it with "as early as v0.5.0"